### DISCUSSION TITLE
suspense and revalidateOnMount are related in an expected way

### DISCUSSION THREAD

**Discussion by GuichiZhao** [vercel/swr#1097] [Q&A]
# Bug report

## Description / Observed Behavior

Data is not fetched when disabling all validation options which is expected. However, data is fetched again when `suspense` is enabled 

## Expected Behavior

validation option and `suspense` should not be related

## Repro Steps / Code Example

```
import { Suspense } from 'react'
import useSWR from 'swr'
export default function Test() {
  return (
    <div>
      <Suspense fallback="loading...">
        <Child1></Child1>
      </Suspense>
      <Suspense fallback="loading...">
        <Child2></Child2>
      </Suspense>
    </div>
  )
}

function Child1() {
  const { data } = useSWR('hello', fetcher, {
    revalidateOnFocus: false,
    revalidateOnMount: false,
    revalidateOnReconnect: false,
    refreshWhenOffline: false,
    refreshWhenHidden: false,
    refreshInterval: 0,
  })
  return <div>child1:{data + ''}</div>
}
function Child2() {
  const { data } = useSWR('world', fetcher, {
    revalidateOnFocus: false,
    revalidateOnMount: false,
    revalidateOnReconnect: false,
    refreshWhenOffline: false,
    refreshWhenHidden: false,
    refreshInterval: 0,
    suspense: true,
  })
  return <div>child2:{data + ''}</div>
}

let i = 0
function fetcher(key) {
  console.log(key, 'invoke fetcher')
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve(++i)
    }, 3000)
  })
}


```## Additional Context

SWR version.  0.4.1

**Comment by koba04** [comment]
@GuichiZhao 
With the `suspense` option, `useSWR` is expected to return data, otherwise, `useSWR` throws a Promise. To throw a Promise, `useSWR` has to invoke a fetcher function. So the current behavior makes sense for me.
Why do you want to avoid invoking a fetcher?

**Comment by shuding** [comment]
This is more like a question, I'll move it to discussion.

When all revalidations are off with suspense enabled, what's your expected `data`? If you don't want to block rendering with data fetching, I can't think of a good reason to enable suspense.
